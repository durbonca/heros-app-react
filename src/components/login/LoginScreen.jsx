import { useContext } from "react"
import { useNavigate } from "react-router-dom"
import { AuthContext } from "../../auth/authContext"
import { types } from "../../types"

export const LoginScreen = () => {

  const navigate = useNavigate()

  const { dispatch } = useContext(AuthContext)
  
  const handleLogin = () => {
    const lastPath = localStorage.getItem('lastPath') || '/'  
    dispatch({ type: types.login , payload: { name: 'usuario' } })
    navigate(lastPath, { replace: true })
  }

  return (
    <div className='container mt-5'>
      <h1>Login</h1>
      <hr />
      <button 
          className='btn btn-primary'
          onClick={handleLogin}
        >
        login
      </button>
    </div>
  )
}
