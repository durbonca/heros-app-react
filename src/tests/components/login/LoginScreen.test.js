import { MemoryRouter, Routes, Route } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import { types } from "../../../types";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { mount } from "enzyme";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe("LoginScreen test", () => {

    const userContext = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }}

    const wrapper = mount(
        <AuthContext.Provider value={userContext}>
            <MemoryRouter initialEntries={['/login']} >
                <Routes>
                    <Route path="/login" element={<LoginScreen />} />
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
        );

        const handleClick = wrapper.find("button").prop('onClick');

    test('should match snapshot', () => { 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("button").text()).toBe('login');
     })

     test('should call dipatch and navigate', () => {
        handleClick();
        // dispatch(...{name: 'usuario'})
        expect(userContext.dispatch).toHaveBeenCalledWith({ 
            type: types.login , 
            payload: { name: 'usuario' } 
        });
        // mockNavigate = (/ ,{repace: true} )
        expect(mockNavigate).toHaveBeenCalledWith("/", { replace: true }); 
     })

     test('should dispatch last route', () => {
        // localstorage.setItem('lastPath', '/dc')
        localStorage.setItem('lastPath', '/dc')     
        // handleClick
        handleClick();
        // mockNavidate... ('dc', {replace: true})
        expect(mockNavigate).toHaveBeenCalledWith("/dc", { replace: true });
     })
})