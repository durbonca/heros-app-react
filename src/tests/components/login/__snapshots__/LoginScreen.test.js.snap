// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen test should match snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
